napi-inl.h

```diff
@@ -5692,6 +5712,53 @@ Addon<T>::DefineProperties(Object object,
 }
 #endif  // NAPI_VERSION > 5
 
+#if NAPI_VERSION > 2
+template <typename Hook, typename Arg>
+Env::CleanupHook<Hook, Arg> Env::AddCleanupHook(Hook hook, Arg* arg) {
+  return CleanupHook<Hook, Arg>(*this, hook, arg);
+}
+
+template <typename Hook>
+Env::CleanupHook<Hook> Env::AddCleanupHook(Hook hook) {
+  return CleanupHook<Hook>(*this, hook);
+}
+
+template <typename Hook, typename Arg>
+Env::CleanupHook<Hook, Arg>::CleanupHook(Napi::Env env, Hook hook)
+    : wrapper(Env::CleanupHook<Hook, Arg>::Wrapper) {
+  data = new CleanupData{std::move(hook), nullptr};
```

Comment:
Could this version just call the other one and pass nullptr for data ?

Reply:
@mhdawson no because each constructor initializes the instance member `wrapper` to a different value, `Env::CleanupHook<Hook, Arg>::Wrapper` (this one) or `Env::CleanupHook<Hook, Arg>::WrapperWithArg`

Reply:
Thanks, nervermind.